doctr/models/recognition/master/pytorch.py

```diff
@@ -0,0 +1,290 @@
+
+        # [N, headers * C', 1, 1]
+        context = context.view(batch, self.headers * self.single_header_inplanes, 1, 1)
+
+        return context
+
+    def forward(self, inputs: torch.Tensor, **kwargs) -> torch.Tensor:
+        # Context modeling: B, C, H, W  ->  B, C, 1, 1
+        context = self.context_modeling(inputs)
+        # Transform: B, C, 1, 1 ->  B, C, 1, 1
+        transformed = self.channel_add_conv(context)
+        return inputs + transformed
+
+
+class MAGCResnet(nn.Sequential):
+
+    """Implements the modified resnet with MAGC layers, as described in paper.
+
+    Args:
+        headers: number of header to split channels in MAGC layers
+        input_shape: shape of the model input (without batch dim)
+    """
+
+    def __init__(
+        self,
+        headers: int = 1,
+    ) -> None:
+        _layers = [
+            # conv_1x
+            *conv_sequence_pt(3, 64, relu=True, bn=True, kernel_size=3, padding=1),
+            *conv_sequence_pt(64, 128, relu=True, bn=True, kernel_size=3, padding=1),
+            nn.MaxPool2d(2),
+            # conv_2x
+            *resnet_stage(128, 256, num_blocks=1),
+            MAGC(inplanes=256, headers=headers, att_scale=True),
+            *conv_sequence_pt(256, 256, relu=True, bn=True, kernel_size=3, padding=1),
+            nn.MaxPool2d(2),
+            # conv_3x
+            *resnet_stage(256, 512, num_blocks=2),
+            MAGC(inplanes=512, headers=headers, att_scale=True),
+            *conv_sequence_pt(512, 512, relu=True, bn=True, kernel_size=3, padding=1),
+            nn.MaxPool2d((2, 1)),
+            # conv_4x
+            *resnet_stage(512, 512, num_blocks=5),
+            MAGC(inplanes=512, headers=headers, att_scale=True),
+            *conv_sequence_pt(512, 512, relu=True, bn=True, kernel_size=3, padding=1),
+            # conv_5x
+            *resnet_stage(512, 512, num_blocks=3),
+            MAGC(inplanes=512, headers=headers, att_scale=True),
+            *conv_sequence_pt(512, 512, relu=True, bn=True, kernel_size=3, padding=1),
+        ]
+        super().__init__(*_layers)
+
+
+class MASTER(_MASTER, nn.Module):
+
+    """Implements MASTER as described in paper: <https://arxiv.org/pdf/1910.02562.pdf>`_.
+    Implementation based on the official Pytorch implementation: <https://github.com/wenwenyu/MASTER-pytorch>`_.
+
+    Args:
+        vocab: vocabulary, (without EOS, SOS, PAD)
+        d_model: d parameter for the transformer decoder
+        headers: headers for the MAGC module
+        dff: depth of the pointwise feed-forward layer
+        num_heads: number of heads for the mutli-head attention module
+        num_layers: number of decoder layers to stack
+        max_length: maximum length of character sequence handled by the model
+        input_size: size of the image inputs
+    """
+
+    def __init__(
+        self,
+        vocab: str,
+        d_model: int = 512,
+        headers: int = 1,
+        dff: int = 2048,
```

Comment:
the arg name is far from being explicit :sweat_smile:

Reply:
It is called that way in the paper :/